src/D2L.CodeStyle.Analyzers/Immutability/MutabilityInspector.cs

```diff
@@ -126,15 +126,47 @@ HashSet<ITypeSymbol> typeStack
 			// sufficiently immutable, and we also need to ensure the
 			// immutability of the type parameters is also appropriate.
 			ImmutabilityScope scope = type.GetImmutabilityScope();
-			if( !type.IsGenericType()
-				&& !flags.HasFlag( MutabilityInspectionFlags.IgnoreImmutabilityAttribute )
+			if( !flags.HasFlag( MutabilityInspectionFlags.IgnoreImmutabilityAttribute )
 				&& scope != ImmutabilityScope.None
 			) {
+				if( type.IsGenericType() ) {
+					// This is some kind of immutable generic class, now we 
+					// need to check that the types specified are appropriate
+					var genericSymbol = type as INamedTypeSymbol;
+					foreach( ITypeSymbol typeArgument in genericSymbol.TypeArguments ) {
+
+						// First, check to see if we care about the immutability
+						// of this argument, otherwise we'll perform an
+						// inspection on a type that doesn't matter.
+						int ordinal = genericSymbol.IndexOfArgument( typeArgument.Name );
+						bool isToBeImmutable = IsTypeArgumentImmutable(
+							genericSymbol.TypeParameters[ordinal],
+							ordinal,
+							genericSymbol );
+
+						if( isToBeImmutable ) {
+
+							MutabilityInspectionResult argumentResult = DoInspectType(
+								typeArgument,
+								type,
+								MutabilityInspectionFlags.AllowUnsealed,
+								typeStack );
+
+							if( argumentResult.IsMutable ) {
+								return argumentResult;
+							}
+						}
+					}
+				}
+
+				// If we didn't bail before here then all the type arguments
+				// are valid, so it's not mutable-ish.
 				ImmutableHashSet<string> immutableExceptions = type.GetAllImmutableExceptions();
 				return MutabilityInspectionResult.NotMutable( immutableExceptions );
 			}
 
 			return DoInspectType(
+				type,

```

Comment:
This doesn't hurt.  We could also easily pass default for the "container" type as well.

Reply:
name these parameters